src/server/scripts/Outland/TempestKeep/botanica/boss_thorngrin_the_tender.cpp

```diff
@@ -17,152 +17,161 @@
     SAY_50_PERCENT_HP           = 4,
     SAY_CAST_HELLFIRE           = 5,
     SAY_DEATH                   = 6,
-    EMOTE_ENRAGE                = 7
+    EMOTE_ENRAGE                = 7,
+    SAY_INTRO                   = 8
 };
 
-enum Spells
+enum ThorngrinSpells
 {
     SPELL_SACRIFICE             = 34661,
     SPELL_HELLFIRE              = 34659,
+    SPELL_HELLFIRE_H            = 39131,
     SPELL_ENRAGE                = 34670
 };
 
-enum Events
+enum ThorngrinEvents
 {
     EVENT_SACRIFICE             = 1,
-    EVENT_HELLFIRE              = 2,
-    EVENT_ENRAGE                = 3
+    EVENT_HELLFIRE,
+    EVENT_ENRAGE
 };
 
-class boss_thorngrin_the_tender : public CreatureScript
+struct boss_thorngrin_the_tender : public BossAI
 {
-    public: boss_thorngrin_the_tender() : CreatureScript("thorngrin_the_tender") { }
+    boss_thorngrin_the_tender(Creature* creature) : BossAI(creature, DATA_THORNGRIN_THE_TENDER), _phase1(false), _phase2(false), _intro(false) { }
+
+    void Reset() override
+    {
+        _Reset();
+        _phase1 = false;
+        _phase2 = false;
+        _intro = false;
+    }
+
+    void MoveInLineOfSight(Unit* who) override
+    {
+        if (!_intro && who->GetTypeId() == TYPEID_PLAYER && me->IsWithinDistInMap(who, 75.0f))
+        {
+            _intro = true;
+            Talk(SAY_INTRO);
+        }
 
-        struct boss_thorngrin_the_tenderAI : public BossAI
+        BossAI::MoveInLineOfSight(who);
+    }
+
+    void JustEngagedWith(Unit* who) override
+    {
+        BossAI::JustEngagedWith(who);
+        Talk(SAY_AGGRO);
+        events.ScheduleEvent(EVENT_SACRIFICE, 30s, 40s);
+        events.ScheduleEvent(EVENT_HELLFIRE, 5s, 15s);
+        events.ScheduleEvent(EVENT_ENRAGE, 30s, 40s);
+    }
+
+    void OnSpellStart(SpellInfo const* spell) override
+    {
+        if (spell->Id == SPELL_SACRIFICE)
+            Talk(SAY_CAST_SACRIFICE);
+    }
+
+    void OnSpellCast(SpellInfo const* spell) override
+    {
+        switch (spell->Id)
         {
-            boss_thorngrin_the_tenderAI(Creature* creature) : BossAI(creature, DATA_THORNGRIN_THE_TENDER)
-            {
-                Initialize();
-            }
+            case SPELL_HELLFIRE:
+            case SPELL_HELLFIRE_H:
+                Talk(SAY_CAST_HELLFIRE);
+                break;
+            case SPELL_ENRAGE:
+                Talk(EMOTE_ENRAGE);
+                break;
+            default:
+                break;
+        }
+    }
 
-            void Initialize()
-            {
-                _phase1 = true;
-                _phase2 = true;
-            }
+    void DamageTaken(Unit* /*killer*/, uint32& damage, DamageEffectType /*damageType*/, SpellInfo const* /*spellInfo = nullptr*/) override
+    {
+        if (me->HealthBelowPctDamaged(50, damage) && !_phase1)
+        {
+            _phase1 = true;
+            Talk(SAY_50_PERCENT_HP);
+        }
+        if (me->HealthBelowPctDamaged(20, damage) && !_phase2)
```

Comment:
can both stuff happen at the same time? if not use `else if`, also i'd suggest to check `!_phase2` first

Reply:
I guess it's possible to have both summons at one time if damage is done fastly enough